pkg/steps/cluster_claim.go

```diff
@@ -99,71 +125,59 @@ func acquireCluster(ctx context.Context, clusterClaim api.ClusterClaim, hiveClie
 	}
 
 	clusterPool := clusterPools.Items[0]
-	claimName := jobSpec.ProwJobID
+	claimName := s.jobSpec.ProwJobID
 	claimNamespace := clusterPool.Namespace
-	claim := &hivev1.ClusterClaim{
+	ret := &hivev1.ClusterClaim{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      claimName,
 			Namespace: claimNamespace,
 			Labels: utils.SanitizeLabels(map[string]string{
-				kube.ProwJobAnnotation: jobSpec.Job,
-				kube.ProwBuildIDLabel:  jobSpec.BuildID,
+				kube.ProwJobAnnotation: s.jobSpec.Job,
+				kube.ProwBuildIDLabel:  s.jobSpec.BuildID,
 			}),
 		},
 		Spec: hivev1.ClusterClaimSpec{
 			ClusterPoolName: clusterPool.Name,
 			Lifetime:        &metav1.Duration{Duration: 4 * time.Hour},
 		},
 	}
-	if err := hiveClient.Create(ctx, claim); err != nil {
-		return nil, fmt.Errorf("failed to created cluster claim %s in namespace %s: %w", claim.Name, claim.Namespace, err)
+	if err := s.hiveClient.Create(ctx, ret); err != nil {
+		return nil, fmt.Errorf("failed to created cluster claim %s in namespace %s: %w", claimName, claimNamespace, err)
 	}
 	logrus.Info("Waiting for the claimed cluster to be ready.")
-	claim = &hivev1.ClusterClaim{}
-	if err := wait.PollImmediate(15*time.Second, clusterClaim.Timeout.Duration, func() (bool, error) {
-		if err := hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: claimName, Namespace: claimNamespace}, claim); err != nil {
-			return false, fmt.Errorf("failed to get cluster claim %s in namespace %s: %w", claim.Name, claim.Namespace, err)
-		}
-		// https://github.com/openshift/hive/blob/a535d29c4d2dc4f7f9bf3f30098c69d9334bee2e/apis/hive/v1/clusterclaim_types.go#L18-L22
-		for _, condition := range claim.Status.Conditions {
-			logrus.WithField("claim.Name", claim.Name).WithField("claim.Namespace", claim.Namespace).
-				WithField("condition.Type", condition.Type).WithField("condition.Status", condition.Status).
-				Debug("Checking the claim's status.")
-			if condition.Type == hivev1.ClusterRunningCondition && condition.Status == corev1.ConditionTrue {
-				return true, nil
-			}
-		}
-		return false, nil
-	}); err != nil {
-		return claim, fmt.Errorf("failed to wait for created cluster claim to become running: %w", err)
+	claim := &hivev1.ClusterClaim{}
+	if err := waitForClaim(s.hiveClient, claimNamespace, claimName, claim, s.clusterClaim.Timeout.Duration); err != nil {
+		return ret, fmt.Errorf("failed to wait for created cluster claim to become ready: %w", err)
 	}
+	ret = claim
 	logrus.Info("The claimed cluster is ready.")
 	clusterDeploymentName := claim.Spec.Namespace
```

Comment:
What's the utility in this local variable?

Reply:
For improve the readability of the code.
Removed.